util/chunk/chunk.go

```diff
@@ -725,3 +725,26 @@ func (c *Chunk) AppendPartialRows(colOff int, rows []Row) {
 		}
 	}
 }
+
+// AppendRowsByColIdxs appends multiple rows by its colIdxs to the chunk.
+// 1. every columns are used if colIdxs is nil.
+// 2. no columns are used if colIdxs is not nil but the size of colIdxs is 0.
+func (c *Chunk) AppendRowsByColIdxs(rows []Row, colIdxs []int) (wide int) {
+	if colIdxs == nil {
+		if len(rows) == 0 {
+			wide = 0
+			return
+		}
+		c.AppendRows(rows)
+		wide = rows[0].Len() * len(rows)
+		return
+	}
+	for _, srcRow := range rows {
+		for i, colIdx := range colIdxs {
+			appendCellByCell(c.columns[i], srcRow.c.columns[colIdx], srcRow.idx)
```

Comment:
`c.appendSel()` is necessary. Rest LGTM.

Reply:
@pingyu Done, PTAL.